src/bidiMapper/domains/script/realm.ts

```diff
@@ -131,22 +105,31 @@ export class Realm {
         // Remember all the handles sent to client.
         this.#realmStorage.knownHandlesToRealm.set(objectId, this.realmId);
       } else {
-        // No need in awaiting for the object to be released.
+        // No need to await for the object to be released.
         void this.#releaseObject(objectId).catch((error) =>
           this.#logger?.(LogType.system, error)
         );
       }
     }
 
+    if (
+      cdpValue.result.type === 'object' &&
+      cdpValue.result.subtype === 'generator'
+    ) {
+      return {
+        type: 'generator',
+      };
+    }
+
     return bidiValue;
   }
 
   deepSerializedToBiDi(
-    webDriverValue: Protocol.Runtime.DeepSerializedValue
+    deepSerializedValue: Protocol.Runtime.DeepSerializedValue
   ): Script.RemoteValue {
     // This relies on the CDP to implement proper BiDi serialization, except
     // backendNodeId/sharedId and `platformobject`.
-    const result = webDriverValue as any;
+    const result = deepSerializedValue as any;
```

Comment:
```

🏃 [tsc] Running command "tsc --build src/tsconfig.json --pretty"
src/bidiMapper/domains/script/realm.ts:135:14 - error TS2339: Property 'internalId' does not exist on type 'DeepSerializedValue'.

135       result.internalId = `${result.weakLocalObjectReference}`;
                 ~~~~~~~~~~

src/bidiMapper/domains/script/realm.ts:141:9 - error TS2367: This comparison appears to be unintentional because the types '"string" | "number" | "bigint" | "boolean" | "symbol" | "undefined" | "object" | "function" | "window" | "array" | "null" | "node" | "regexp" | "date" | "map" | "set" | "weakmap" | ... 5 more ... | "arraybuffer"' and '"platformobject"' have no overlap.

141     if (result.type === 'platformobject') {
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/bidiMapper/domains/script/realm.ts:147:7 - error TS2322: Type 'DeepSerializedValue' is not assignable to type 'RemoteValue'.
  Type 'DeepSerializedValue' is not assignable to type 'StringValue | NumberValue | BooleanValue | BigIntValue | ArrayRemoteValue | ObjectRemoteValue | ... 7 more ... | WindowProxyRemoteValue'.
    Type 'DeepSerializedValue' is not assignable to type 'WindowProxyRemoteValue'.
      Types of property 'type' are incompatible.
        Type '"string" | "number" | "bigint" | "boolean" | "symbol" | "undefined" | "object" | "function" | "window" | "array" | "null" | "node" | "regexp" | "date" | "map" | "set" | "weakmap" | ... 5 more ... | "arraybuffer"' is not assignable to type '"window"'.
          Type '"string"' is not assignable to type '"window"'.

147       return result;
          ~~~~~~

src/bidiMapper/domains/script/realm.ts:153:16 - error TS2339: Property 'sharedId' does not exist on type 'DeepSerializedValue'.

153         result.sharedId = `${this.navigableId}${SHARED_ID_DIVIDER}${bidiValue.backendNodeId}`;
                   ~~~~~~~~

src/bidiMapper/domains/script/realm.ts:190:5 - error TS2322: Type 'DeepSerializedValue' is not assignable to type 'RemoteValue'.

190     return result;
        ~~~~~~


```

Reply:
I'll remove the assignment and typecast where it's needed.